### PR TITLE
hancock: `disable` as it stops working

### PR DESCRIPTION
**Pull request by chenrui333**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---

- https://github.com/JeremyAgost/Hancock/issues/8
- https://github.com/JeremyAgost/Hancock/issues/7
- https://github.com/JeremyAgost/Hancock/issues/6

### CODE REVIEW

**Comment by krehel** [comment]
If the app no longer works it should be removed imho

**Comment by SMillerDev** [comment]
We can disable it then, which would make for a much nicer experience for users.

**Comment by chenrui333** [comment]
> If the app no longer works it should be removed imho

I think it just does not work for the newer systems (that is my guess)

**Comment by krehel** [comment]
That's fair.  I tend to err on not keeping broken things, especially if non functional for a relatively long time.  `disable` is a good option.

**Comment by SMillerDev** [comment]
Not letting people install broken things is good, but if you remove the cask existing users will never know they should look for a replacement.

**Comment by krehel** [comment]
Great point @SMillerDev.  Always appreciate your insights.  I am probably too aggressive with my thought process, the mediation is good. 👍

**Review by bevanjkay — APPROVED** [review]

File: Casks/h/hancock.rb [hunk 1]
```diff
@@ -6,5 +6,12 @@
   name "Hancock"
   homepage "https://github.com/JeremyAgost/Hancock"
 
+  # it has been four years since the last release
+  # users have reported it stopped working for m1 as well as osx
+  # https://github.com/JeremyAgost/Hancock/issues/8
+  # https://github.com/JeremyAgost/Hancock/issues/7
+  # https://github.com/JeremyAgost/Hancock/issues/6
+  disable! date: "2024-08-01", because: :unmaintained
```
Comment (bevanjkay) [inline Casks/h/hancock.rb:14]: @chenrui333 How is the date determined for disabling?

**Review by bevanjkay — CHANGES_REQUESTED** [review]

File: Casks/h/hancock.rb (hunk 1 above)
Comment (p-linnane) [reply Casks/h/hancock.rb:14]: This would be a good AGM topic, as we are planning to discuss how we do it for formulae then.

Comment (chenrui333) [reply Casks/h/hancock.rb:14]: that is a typo, should be 2024-01-01 😆

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**